infra/main.bicep

```diff
@@ -9,48 +9,27 @@ param name string
 @description('Primary location for all resources')
 param location string
 
-@secure()
-@description('PostGreSQL Server administrator password')
-param postgresServerPassword string
-
 @description('Id of the user or app to assign application roles')
 param principalId string = ''
 
+@secure()
+@description('PostGreSQL Server administrator password')
+param postgresAdminPassword string
+
 var resourceToken = toLower(uniqueString(subscription().id, name, location))
 var tags = { 'azd-env-name': name }
 var prefix = '${name}-${resourceToken}'
 
-var postgresServerName = '${prefix}-postgres'
-var postgresServerAdmin = 'flaskadmin'
-var postgresDatabaseName = 'transtel'
-
 resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
   name: '${prefix}-rg'
   location: location
   tags: tags
 }
 
-// Store secrets in a keyvault
-module keyVault './core/security/keyvault.bicep' = {
```

Comment:
What happened to key vault?

Reply:
It was not being used initially so, I removed it.
The db password was being defined and stored there but not used by db from the keyvault.

Tried to utilize the key vault in storing the cognitive service key found it's not possible as the key has to be auto created by service and then changes to customer managed key later but during the creation it can't be set.

so, removed it as I found no reason for having it but I can re add it if you want.

Reply:
The KV gets used by secretOrRandomPassword- once something gets stored in there, it will be retrieved from there on subsequent deploys. Otherwise secretOrRandomPassword will keep generating new passwords each time.

Reply:
Ohhh, didn't know that.
I will add it again.

Reply:
Re storing cog service key in KV- I haven't done that myself yet but Jon Gallant is working on a PR that does it, see:
https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2/files#diff-7dc4bd7866595a4b8b38b2fe17f1770bd7c2e969867143dae801a3f2e673846aR17

Is that similar to what you tried?

Reply:
Nope and their approach will work.
They are not enforcing a generated random key from the bicep on the cognitive service.

They are retrieving the key the same way I'm doing after the service is created and saving it into the key vault instead of application setting.

so, what they are doing is doable but currently I'm not fully aware of the pros and cons of their approach.

will apply your review tomorrow's morning.

and look more into this after readding the KV